Lib/Rabbit4000/BIOSLIB/MUTIL.LIB

```diff
@@ -351,48 +712,77 @@ L_asr::
 ; right shift unsigned item on stack by count in bcde
 #endasm
 
-/*** beginheader G_asr */
+/*** beginheader G_asr, _fast_asr */
 // signed & unsigned long bit operations
 __root void G_asr();
+__root void _fast_asr();
 /*** endheader */
 #asm __nodebug
+   align even
 G_asr::  ; almost the same as L.asr, except sign is not preserved
 	pop   iy ;return
 	; In practice, shifts by 1, 8, 16 or 24 would be much more common than any others.
    ; But we don't take advantage of this, since the compiler should in-line constant shifts.
-	ld		jkhl,31
+   ld		jkhl,31
    cp		jkhl,bcde
-	jp		c,_ZGLExit
+   jp		c,_ZGLExit
    ld		a,e			; A <= 31.
-	pop	bcde
-   rra
-   jr		nc,@pc+4
+   pop	bcde
+   rr a
+   jr		nc,@pc+5
    srl	1,bcde
-   rra
-   jr		nc,@pc+4
+   rr a
+   jr		nc,@pc+5
    srl	2,bcde
-   rra
-   jr		nc,@pc+4
+   rr a
+   jr		nc,@pc+5
    srl	4,bcde
-	rra
-   jr		nc,@pc+6
+   rr a
+   jr		nc,@pc+7
    rrc	8,bcde
    ld		B,0
-   rra
+   rr a
    jr		nc,@pc+5
    clr	hl
    ex		bc,hl
    ex		de,hl
    jp		(iy)
+
+; Speedier version with no stack args, value in bcde and a set to shift (<= 31)
+; useful for maths routines with limited ranges
+   align even
+_fast_asr::
+   rr a
+   jr		nc,@pc+5

```

Comment:
I realize the original code used the `@pc+x` notation, but seeing it here feels risky -- it depends on knowing exactly how many opcodes show up in following statements.  Is there a reason we can't just replace the `jr` destinations with local labels?